### PR TITLE
Add table oci_identity_db_credential Closes #594

### LINKED ISSUE #594: Add table oci_identity_db_credential
**References**
https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/identity_db_credential

Note: This is required for CIS V2.0.0

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_identity_db_credential' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_identity_db_credential []

PRETEST: tests/oci_identity_db_credential

TEST: tests/oci_identity_db_credential
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # oci_identity_db_credential.named_test_resource will be created
  + resource "oci_identity_db_credential" "named_test_resource" {
      + description       = "testing"
      + id                = (known after apply)
      + lifecycle_details = (known after apply)
      + password          = (sensitive value)
      + state             = (known after apply)
      + time_created      = (known after apply)
      + time_expires      = (known after apply)
      + user_id           = (known after apply)
    }

  # oci_identity_user.named_test_resource will be created
  + resource "oci_identity_user" "named_test_resource" {
      + capabilities                   = (known after apply)
      + compartment_id                 = "ocid1.tenancy.oc1..eeeeeeeeueweljrlncnweuedhjsadehkjebcbehbeuyhshsshekkkekekjff"
      + db_user_name                   = (known after apply)
      + defined_tags                   = (known after apply)
      + description                    = "steampipetest304"
      + email                          = (known after apply)
      + email_verified                 = (known after apply)
      + external_identifier            = (known after apply)
      + freeform_tags                  = (known after apply)
      + id                             = (known after apply)
      + identity_provider_id           = (known after apply)
      + inactive_state                 = (known after apply)
      + last_successful_login_time     = (known after apply)
      + name                           = "steampipetest304"
      + previous_successful_login_time = (known after apply)
      + state                          = (known after apply)
      + time_created                   = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + credential_id   = (known after apply)
  + description     = "testing"
  + lifecycle_state = (known after apply)
  + tenancy_ocid    = "ocid1.tenancy.oc1..eeeeeeeeueweljrlncnweuedhjsadehkjebcbehbeuyhshsshekkkekekjff"
  + user_id         = (known after apply)
oci_identity_user.named_test_resource: Creating...
oci_identity_user.named_test_resource: Creation complete after 2s [id=ocid1.user.oc1..aaaaaaaak2bgsaa47k5kkfojmsapmci3etnueefbzbyxb2x4oswthacmqorq]
oci_identity_db_credential.named_test_resource: Creating...
oci_identity_db_credential.named_test_resource: Creation complete after 1s [id=users/ocid1.user.oc1..aaaaaaaak2bgsaa47k5kkfojmsapmci3etnueefbzbyxb2x4oswthacmqorq/dbCredentials/ocid1.credential.oc1..aaaaaaaayxcy6yjqx3qiu5dvswifsmu7ua6jij7bzj6xzu4ezvegojyhusuq]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

credential_id = "ocid1.credential.oc1..aaaaaaaayxcy6yjqx3qiu5dvswifsmu7ua6jij7bzj6xzu4ezvegojyhusuq"
description = "testing"
lifecycle_state = "ACTIVE"
tenancy_ocid = "ocid1.tenancy.oc1..eeeeeeeeueweljrlncnweuedhjsadehkjebcbehbeuyhshsshekkkekekjff"
user_id = "ocid1.user.oc1..aaaaaaaak2bgsaa47k5kkfojmsapmci3etnueefbzbyxb2x4oswthacmqorq"

Running SQL query: test-list-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)

[
  {
    "description": "testing",
    "id": "ocid1.credential.oc1..aaaaaaaayxcy6yjqx3qiu5dvswifsmu7ua6jij7bzj6xzu4ezvegojyhusuq",
    "lifecycle_state": "ACTIVE",
    "user_id": "ocid1.user.oc1..aaaaaaaak2bgsaa47k5kkfojmsapmci3etnueefbzbyxb2x4oswthacmqorq"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)

[]
✔ PASSED

Running SQL query: test-turbot-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)

[
  {
    "tenant_id": "ocid1.tenancy.oc1..eeeeeeeeueweljrlncnweuedhjsadehkjebcbehbeuyhshsshekkkekekjff",
    "title": "ocid1.credential.oc1..aaaaaaaayxcy6yjqx3qiu5dvswifsmu7ua6jij7bzj6xzu4ezvegojyhusuq"
  }
]
✔ PASSED

POSTTEST: tests/oci_identity_db_credential

TEARDOWN: tests/oci_identity_db_credential

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from oci_identity_db_credential
+------------------------------------------------------------------------------------+------------------------------------------------------------------------------+-------------+---------------------------+---------------------------+-----------------+---------------->
| id                                                                                 | user_id                                                                      | description | time_created              | time_expires              | lifecycle_state | title          >
+------------------------------------------------------------------------------------+------------------------------------------------------------------------------+-------------+---------------------------+---------------------------+-----------------+---------------->
| ocid1.credential.oc1..aaaaaaaaft6oq4slulmxpolyrhq6v4zs53gqfj4kj7wrfn2ssi6jd7xvgqia | ocid1.user.oc1..aaaaaaaa2jjigunx3y76rdbx6acvv4kueepbh6hnmr2yr4wnzyuyqi2jaz6a | <null>      | 2024-01-25T12:18:23+05:30 | 2024-01-25T12:18:23+05:30 | ACTIVE          | ocid1.credentia>
+------------------------------------------------------------------------------------+------------------------------------------------------------------------------+-------------+---------------------------+---------------------------+-----------------+---------------->
```
</details>

### CODE REVIEW

File: docs/tables/oci_identity_db_credential.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+title: "Steampipe Table: oci_identity_db_credential - Query OCI Identity DB Credential using SQL"
+description: "Allows users to query information about users DB credential in OCI Identity."
+---
+
+# Table: oci_identity_db_credential - Query OCI Identity Users using SQL
+
+Oracle Cloud Infrastructure's Identity and Access Management (IAM) service lets you control who has access to your cloud resources. Identity DB credentials refer to the authentication details used to access Oracle databases securely within the Oracle Cloud Infrastructure. These credentials are part of the Oracle Cloud Infrastructure Identity and Access Management (IAM) service, which manages users, groups, and policies that control access to OCI resources. In the context of databases, these credentials
```
Comment (khushboo9024) [inline docs/tables/oci_identity_db_credential.md:8]: @ParthaI Is this description completed?

File: docs/tables/oci_identity_db_credential.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+select
+  name,
+  id,
+  user_id,
+  description,
+  time_created,
+  lifecycle_state,
+  time_expires
+from
+  oci_identity_db_credential;
```
Comment (khushboo9024) [inline docs/tables/oci_identity_db_credential.md:29]:
```suggestion
select
  id,
  user_id,
  description,
  time_created,
  lifecycle_state,
  time_expires
from
  oci_identity_db_credential;
```

File: docs/tables/oci_identity_db_credential.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
```
Comment (khushboo9024) [inline docs/tables/oci_identity_db_credential.md:42]:
```suggestion
select
  id,
  user_id,
  description,
  time_created,
  lifecycle_state,
  time_expires
from
  oci_identity_db_credential;
```

File: docs/tables/oci_identity_db_credential.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  c.db_credential_id,
+  u.user_type,
+  u.is_mfa_activated,
+  u.email
+from
+  oci_identity_db_credential as c,
+  oci_identity_user as u
+where
+  c.user_id = u.user_id
+  and user_type = 'IDCS';
```
Comment (khushboo9024) [inline docs/tables/oci_identity_db_credential.md:60]:
```suggestion
select
  c.id,
  u.user_type,
  u.is_mfa_activated,
  u.email
from
  oci_identity_db_credential as c,
  oci_identity_user as u
where
  c.user_id = u.id
  and user_type = 'IDCS;
```

File: docs/tables/oci_identity_db_credential.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  c.user_id,
+  c.db_credential_id,
+  u.user_type,
+  u.is_mfa_activated,
+  u.email
+from
+  oci_identity_db_credential as c
+  join oci_identity_user as u on c.user_id = u.user_id
+where
+  u.user_type = 'IDCS';
```
Comment (khushboo9024) [inline docs/tables/oci_identity_db_credential.md:74]:
```suggestion
select
  c.id,
  u.user_type,
  u.is_mfa_activated,
  u.email
from
  oci_identity_db_credential as c
  join oci_identity_user as u on c.user_id = u.id
where
  u.user_type = 'IDCS';
```

File: docs/tables/oci_identity_db_credential.md (hunk 1 above)
Comment (ParthaI) [reply docs/tables/oci_identity_db_credential.md]: Updated.

File: docs/tables/oci_identity_db_credential.md (hunk 2 above)
Comment (ParthaI) [reply docs/tables/oci_identity_db_credential.md]: Updated

File: docs/tables/oci_identity_db_credential.md (hunk 3 above)
Comment (ParthaI) [reply docs/tables/oci_identity_db_credential.md]: Updated the query

File: docs/tables/oci_identity_db_credential.md (hunk 4 above)
Comment (ParthaI) [reply docs/tables/oci_identity_db_credential.md]: Updated

File: docs/tables/oci_identity_db_credential.md (hunk 5 above)
Comment (ParthaI) [reply docs/tables/oci_identity_db_credential.md]: Updated

**Review by khushboo9024 — APPROVED** [review]

**Pull request merged**